pulsar-client-cpp/lib/HandlerBase.cc

```diff
@@ -48,11 +48,13 @@ void HandlerBase::start() {
 }
 
 void HandlerBase::grabCnx() {
+    Lock lock(mutex_);
     if (connection_.lock()) {
+        lock.unlock();
```

Comment:
Agreed but in previous PRs I have got a comment suggesting to do IO operations outside the log - hence the unlock